flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Executing.java

```diff
@@ -144,6 +188,17 @@ private void maybeRescale() {
         }
     }
 
+    private Duration timeSinceLastRescale() {
+        return Duration.between(lastRescale, Instant.now());
+    }
+
+    private void rescaleWhenCooldownPeriodIsOver() {
```

Comment:
`rescaleWhenCooldownPeriodIsOver` should consider one case that it's called in multiple times.

For example, scalingIntervalMin is 30 seconds, `lastRescale` is 08:01:05.

- The `rescaleWhenCooldownPeriodIsOver` is called at  08:01:10, it will schedule `maybeRescale` at `08:01:35`.
- The `rescaleWhenCooldownPeriodIsOver` maybe called at  08:01:15 again, it will schedule `maybeRescale` at `08:01:35` again.

The second or repeated schedule should be ignored.

Reply:
ok thanks for pointing out.

Reply:
If we don't drop it or don't have any filter, the `scaling-interval.min` cannot be guaranteed, right?

The interval of  `08:01:40` and `08:01:45` is 5 s, and the `scaling-interval.min`  is 30s. It may not be as expected. Do you think so?

Reply:
Exactly. This is what I meant and implemented in last commit of yesterday : the call at 08:01:10 will schedule `maybeRescale` at 08:01:40 and the call at 08:01:15 will not entail any rescale. When `maybeRescale` is executed at  08:01:40, it will use the resources added at 08:01:15 when calling `Context#getExecutionGraph()`. So that seems fine.